### ISSUE TITLE
Heredoc formatting

### ISSUE THREAD

**Issue by knyttl** [pinterest/ktlint#1360]
## Expected Behavior
I would expect this code to be properly formatted:
```kotlin
val foo =
    """
        ${
            if (1 == 1)
                1
            else 
                2
        }
    """.trimIndent()
```
## Observed Behavior
ktlint prefers following:

```kotlin
val foo =
    """
        ${
    if (1 == 1)
        1
    else
        2
    }
    """.trimIndent()
```

## Steps to Reproduce
Just let ktlint format the code above. Also observed in Idea with ktlint plugin:

<img width="424" alt="Screenshot 2022-02-04 at 12 24 14" src="https://user-images.githubusercontent.com/401107/152521017-8f2da205-072d-4e1c-b92d-cd200cbffd05.png">

## Your Environment
```
gradle:
id("org.jlleitschuh.gradle.ktlint") version "10.2.1"

ktlint plugin:
0.8.2

* Version of ktlint used: 0.43.2
* Version of Gradle used (if applicable): 3.7.1
* Operating System and version: Any

**Comment by paul-dingemans** [comment]
The code is formatted in same way as the default IntellI IDEA Formatter:
```
val foo =
    """
        ${
        if (1 == 1)
            1
        else
            2
    }
    """.trimIndent()
```

If we would use another formatting style like the one you expected (which undoubtly is better) than we would produce code that would be rejected/reformatted by the IntelliJ IDEA formatter. For now, we still follow the rule that code produced by ktlint should not conflict with the IntelliJ IDEA formatter even in case it results in ugly code.

The problem in this case is cause by the indent before opening of the Heredoc. And there is a good reason for that. Suppose that we would exagarrate that indent to:
```
val foo =
    """
        some looooooooooooooooooooon text ${
        if (1 == 1)
            1
        else
            2
    }
    """.trimIndent()
```
There is no way that we can format this properly in a consistent way with what you expect.
